Update serialization documentation (#19732)
metabase/metabase: docs/enterprise-guide/serialization.md

- Saved questions

> - Pulses

Reviewer: ```suggestion
```

- User accounts or settings
- Alerts on saved questions
- Personal Collections or their contents (except for the user specified with the `--user` flag; see below)

> - Revision history of questions and dashboards (so questions and dashboards loaded via serialization won't have "last edited by" and "last edited at" property)

Reviewer 2: ```suggestion
```
Again, I don't think I docs should list all the known bugs - otherwise we should have 600+ additions to our docs.
Something that are clear limitations should of course be listed, but not bugs.
And this one has always been a problem, but just never got noticed, since this metadata wasn't very prominent until lately.

`java -jar metabase.jar load [my_dump] --mode [skip/update] --on-error [continue/abort]`

> The `--mode` flag lets you specify what to do when encountering a duplicate dashboard, question, or any Admin Panel settings that already set (again, except for permissions and user settings, which are not currently included in data dumps). It can either `skip` that item and do nothing to it, or `update` it with the version being loaded. The default is `skip`. For example, if you change the order, add or remove any cards in a dashboard that has been loaded previously, mode `skip` won't do any modifications to it, as the load process "skips" any object that previously exists no matter if there are changes on the source file.

Reviewer: ```suggestion
The `--mode` flag lets you specify what to do when encountering a duplicate dashboard, question, or any Admin Panel settings that already set (again, except for permissions and user settings, which are not currently included in data dumps). It can either `skip` that item and do nothing to it, or `update` it with the version being loaded. The default is `skip`. For example, even if you change the card order for a dashboard, or add or remove any cards in a dashboard, the `skip` mode won't modify the dashboard object, as the load process "skips" any object that previously existed, regardless of changes in the source file.
```
Does this revised text capture everything?

Reviewer 3: perfect

The `--on-error` flag allows you to specify whether the load process should keep going or stop when there's an error. The default is `continue`. Note that `abort` won't undo any successful artifact loads that happened before an error was encountered.

Both of these flags are optional.

### Current limitations

> - The serialization feature is not able to identify a card in the destination if the visualization settings in the source have changed

Reviewer 2: That doesn't sound correct. Might be applicable to dashcards, but not cards. And what does "not able to identify" mean?
If this is one of the new issues, then it's not a limitation, but a bug and should not be listed in the docs.

Reviewer 3: Correct, it's a dashcard. Agree that it's a bug, but as it's a difficult one and might not be fixed in the short term, I wanted to raise some red flag here so users can take this into consideration when using the feature.

Reviewer 2: There's currently 74 other Correctness issues, we don't list those in the docs - and shouldn't. Then our docs would be filled with "limitations".

> - The serialization feature is not able to identify a dashboard card in the destination if the visualization settings in the source dashboard have changed

Reviewer: Is this what you mean?

```suggestion
-  If the only change for a dashboard card is its visualization settings (that is, the visualization settings on the dashboard card, not the visualization for the card's question), the serialization process won't pick up that change, which means the serialization process won't apply the card's new dashboard visualization setting to the dashboard card at the destination.
```

And is there a workaround?

Reviewer 3: in fact it will duplicate the card on destination, but Johannes does not agree with adding this section so we should delete it
